src/pd_gmres.m

```diff
@@ -97,32 +123,60 @@
 
 clear rowsb colsb;
 
-% ----> Default values and sanity checks for parameter m0
-
+% ----> Default values and sanity checks for parameter mInitial
 % When the restart parameter is not specified explicitly as an input
 % argument, pd_gmres() behaves identically to the unrestarted gmres()
 % with a fixed number of iterations given by min(n, 10).
 %
 % There are three possibilities when this might happen: 
 %   (1) If only two arguments are given i.e., A and b.
-%   (2) If m0 equals the dimension of A i.e., m0 = n.
-%   (3) If an empty matrix is given as restart parameter, i.e., m0 = [].
-if (nargin < 3) || isempty(m0) || (m0 == n)
+%   (2) If mInitial equals the dimension of A i.e., mInitial = n.
+%   (3) If an empty matrix is given as mInitial, i.e., mInitial = [].
+if (nargin < 3) || isempty(mInitial) || (mInitial == n)
     restarted = false;  % use unrestarted pd_gmres()
 else
     restarted = true;  % use restarted pd_gmres()
 end
 
-if restarted
-    if m0 <= 0
-        error("Restart parameter m0 must be a positive integer.")
-    elseif m0 > n
-        error("Restart parameter m0 cannot be greater than n.")
+if restarted && (mInitial <= 0) || (mInitial > n)
+    error("mInitial must satisfy: 0 < mInitial <= n.")
+end
+
+% ----> Default values and sanity checks for mMinMax
+if (nargin < 4) || isempty(mMinMax)
+    mMinMax = [1; n-1];
+else
+    if ~restarted
+        warning("mMinMax was given but will not be used.")
     end
 end
 
+% TODO: Do we want to allow for mMinMax(1) = mMinMax(2)? I guess in that
+% case the PD-rule won't do anything, since m will remain constant. Right?
+if (mMinMax(1) <= 0) || (mMinMax(2) > n) || (mMinMax(2) <= mMinMax(1))
+    error("mMinMax must satisfy: 0 < mMinMax(1) < mMinMax(2) <= n.")
+end
+
+mMin = mMinMax(1);
+mMax = mMinMax(2);
+
+% ----> Default values and sanity checks for mStep
+if (nargin < 5) || isempty(mStep)
+    mStep = 1;
+else
+    if ~restarted
+        warnign("mStep was given but will not be used.")
+    end
+end
+   
+%TODO: Do we want to allow for mStep = 0? In that case, the m parameter

```

Comment:
@jccf19 Please have a look at this question.

Reply:
Should be  mStep>0.